model/src/main/scala/ch.epfl.scala.index.model/release/ScalaTarget.scala

```diff
@@ -37,86 +41,92 @@ sealed trait ScalaTarget extends Ordered[ScalaTarget] {
+  val encode = s"_sjs${scalaJsVersion}_$backend"
   val targetType: ScalaTargetType = Js
-  val isValid: Boolean = ScalaJvm.isValid(scalaVersion) && ScalaJs.isValid(
+  val isValid: Boolean = backend.isValid && ScalaJs.isValid(
     scalaJsVersion
   )
 }
-case class ScalaNative(scalaVersion: BinaryVersion,
+case class ScalaNative(backend: BackendVersion,
                        scalaNativeVersion: BinaryVersion)
     extends ScalaTarget {
-  val render = s"scala-native $scalaNativeVersion ($scalaVersion)"
-  val encode = s"_native${scalaNativeVersion}_$scalaVersion"
+  val render = s"scala-native $scalaNativeVersion (${backend.render})"
+  val encode = s"_native${scalaNativeVersion}_$backend"
   val targetType: ScalaTargetType = Native
-  val isValid: Boolean = ScalaJvm.isValid(scalaVersion) && ScalaNative.isValid(
+  val isValid: Boolean = backend.isValid && ScalaNative.isValid(
     scalaNativeVersion
   )
 }
 
-case class SbtPlugin(scalaVersion: BinaryVersion, sbtVersion: BinaryVersion)
+case class SbtPlugin(backend: BackendVersion, sbtVersion: BinaryVersion)
     extends ScalaTarget {
-  val render = s"sbt $sbtVersion ($scalaVersion)"
-  val encode = s"_${scalaVersion}_$sbtVersion"
+  val render = s"sbt $sbtVersion ($backend)"
+  val encode = s"_${backend}_$sbtVersion"
   val targetType: ScalaTargetType = Sbt
-  val isValid: Boolean = ScalaJvm.isValid(scalaVersion) && SbtPlugin.isValid(
+  val isValid: Boolean = backend.isValid && SbtPlugin.isValid(
     sbtVersion
   )
 }
 
 object ScalaJvm {
-  def fromFullVersion(fullVersion: SemanticVersion): ScalaJvm = {
+  def fromFullVersion(fullVersion: SemanticVersion): Option[ScalaJvm] = {
     val binaryVersion = fullVersion match {
-      case SemanticVersion(major, minor, patch, _, Some(preRelease), _) =>
-        PreReleaseBinary(
-          major,
-          minor.getOrElse(0),
-          patch,
-          preRelease
-        )
-      case SemanticVersion(major, minor, _, _, _, _) =>
-        MinorBinary(major, minor.getOrElse(0))
+      case SemanticVersion(major, Some(minor), _, None, None, None) =>
+        Some(MinorBinary(major, minor))
+      case _ => None
     }
-    ScalaJvm(binaryVersion)
-  }
-
-  val `2.10`: BinaryVersion = MinorBinary(2, 10)
-  val `2.11`: BinaryVersion = MinorBinary(2, 11)
-  val `2.12`: BinaryVersion = MinorBinary(2, 12)
-  val `2.13`: BinaryVersion = MinorBinary(2, 13)
 
-  private val stableBinaryVersions = Set(`2.10`, `2.11`, `2.12`, `2.13`)
+    binaryVersion
+      .collect {
+        case version if ScalaBackend.isValid(version) => ScalaBackend(version)
+        case version if DottyBackend.isValid(version) => DottyBackend(version)
+      }
+      .map(ScalaJvm(_))
+  }
 
-  def isValid(version: BinaryVersion): Boolean =
-    stableBinaryVersions.contains(version)
+  def Parser[_: P]: P[ScalaTarget] =
+    ("_" ~ BackendVersion.Parser).map(ScalaJvm.apply)
 }
 
-object ScalaJs {
+object ScalaJs extends Parsers {
```

Comment:
Why is this necessary? `ScalaNative` below doesn't have it.

Reply:
it is not :)